### ISSUE TITLE
Moving files in and out of subdirectory in git project gives "Failed to move entry" and breaks git and GitHub integration 

### ISSUE THREAD

**Issue by KittyKittyKitKat** [atom/atom#22532]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description
Upon moving files in and out of a subdirectory in a git project, Atom flashes a pop-up detailing that it failed to move the entry from the starting path to the new path, with the error message of "Cannot read property getPath of undefined". Despite this, the files are still moved both on disk and in the file pane. This error causes git and GitHub integration within Atom to fail. The files being moved are not detected properly as changed in the git pane of Atom. (The changes appear to still be properly recognized by git in the command line interface.) Rebooting Atom causes the changes to be recognized in the git pane, but the GitHub pane is stuck on loading. A final reboot of Atom causes Atom to fail to recognize the repository at all, with the error message present throughout this whole process.

The final result of this is that the git project is entirely unusable within Atom.

### Steps to Reproduce
1. Boot Atom into safe mode (`$ atom --safe`)
2. Move one or more files into or out of a subdirectory (new or existing) within a git project.
3. Observe the error message popup and notice that Atom's git and GitHub integration fails with multiple reboots doing the same actions.

The following video, screen recorded from my machine, clearly illustrates the exact behaviour that caused this error.

https://user-images.githubusercontent.com/57276022/120517507-88a9a380-c39e-11eb-8db0-339ea27da9dd.mp4

**Expected behavior:**
Files to be moved without errors, git integration to recognize moved files when they are moved, and GitHub integration to not fail to connect to a remote repository after these errors occur.

**Actual behavior:**
Moving files throws an error message, and git and GitHub integration fails, rendering the entire project unusable from Atom.

**Reproduces how often:**
Moving files throws an error, and git and GitHub integration fails. This occurs every time a file is moved in a git project within Atom, and the only way I have been able to restore functionality is by clone the GitHub repo to a new project in Atom, or backing up the project and use that.

### Versions

```
Atom    : 1.57.0
Electron: 9.4.4
Chrome  : 83.0.4103.122
Node    : 12.14.1
```
OS Version: Ubuntu (Debian) 20.04.2 LTS Focal Fossa

### Additional Information
* ~~This issue was not present when reverting back to Atom version 1.56.0 and this issue is not present in projects that are not git projects; moving files works as expected.~~ See update comment
* Atom was run with a single monitor setup (Lenovo Ideapad5) with the standard US keyboard layout (QWERTY).
* No configuration files are used.
* This is my current list of packages, although this issue was present in both normal operation and safe modes.
```
Community Packages (21)
├── MagicPython@1.1.1
├── atom-jinja2@0.6.0
├── atom-live-server@2.3.0
├── brunel-syntax@0.1.0
├── busy-signal@2.0.1
├── emmet@2.4.3
├── file-icons@2.1.47
├── hex-view@1.0.1
├── ide-json@0.2.1
├── intentions@1.1.5
├── kite@0.205.0
├── language-brainfuck@0.2.0
├── language-x86-64-assembly@2.3.0
├── linter@3.3.1
├── linter-flake8@2.4.0
├── linter-ui-default@3.3.1
├── minimap@4.39.14
├── pigments@0.40.6
├── platformio-ide-terminal@2.10.1
├── python-autopep8@0.1.3
└── python-docstring-templates@0.2.0
```

**Comment by KittyKittyKitKat** [comment]
Update: Further testing with Atom version 1.56.0 revealed that moving files into a folder did not throw an error but still results in the same breakage of GitHub and git

**Comment by icecream17** [comment]
Can you show the error message? <sup>`atom --dev` gets better stack traces in the console. Maybe this isn't logged in the console, but wanted to say this.</sup>

**Comment by KittyKittyKitKat** [comment]
Running with `atom --dev` and checking the console didn't yield anything blatantly related to the error thrown. The 2 actuall errors I have are
```
Uncaught (in promise) electron/js2c/renderer_init.js:131
Error: ERR_FILE_NOT_FOUND (-6) loading 'file:///usr/share/atom/resources/app.asar.unpacked/node_modules/github/lib/renderer.html?js=%2Fusr%2Fshare%2Fatom%2Fresources%2Fapp.asar.unpacked%2Fnode_modules%2Fgithub%2Flib%2Fworker.js&managerWebContentsId=1&operationCountLimit=10&channelName=github%3Arenderer-ipc'
    at rejectAndCleanup (electron/js2c/browser_init.js:225:1288)
    at WebContents.failListener (electron/js2c/browser_init.js:225:1505)
    at WebContents.emit (events.js:223:5)
```
and
```
Uncaught (in promise) /home/nyxian/.atom/packages/linter-ui-default/node_modules/atom-package-deps/lib/index.js:553 
Error: Dependency#2 for linter-ui-default is invalid
    at invariant (/home/nyxian/.atom/packages/linter-ui-default/node_modules/atom-package-deps/lib/index.js:553)
    at /home/nyxian/.atom/packages/linter-ui-default/node_modules/atom-package-deps/lib/index.js:571
    at Array.forEach (<anonymous>)
    at getDependencies$2 (/home/nyxian/.atom/packages/linter-ui-default/node_modules/atom-package-deps/lib/index.js:561)
    at install (/home/nyxian/.atom/packages/linter-ui-default/node_modules/atom-package-deps/lib/index.js:912)
```
The rest of my console is filled with warnings about "DevTools failed to load source map" (17 of those to be exact)

**Comment by KittyKittyKitKat** [comment]
Here's an interesting update after fiddling around and trying different combinations of interactions with Atom. To summarize the video, I moved the files around in the same manner, still getting the pop-up message. Through repeated uses of `git status` and `git add` I was able to coax the git integration on Atom to register the changes being made and prevent total breakage. I still think this is an issue; Atom should be able to recognize these things. But maybe this is more of a git integration issue than just a file moving issue.

https://user-images.githubusercontent.com/57276022/120537853-c6fd8d80-c3b3-11eb-9388-ceb9b2db9933.mp4
